diff --git a/init/init.c b/init/init.c
index f4855e2..084e585 100755
--- a/init/init.c
+++ b/init/init.c
@@ -683,6 +683,7 @@ static void export_kernel_boot_props(void)
         { "ro.boot.mode", "ro.bootmode", "unknown", },
         { "ro.boot.baseband", "ro.baseband", "unknown", },
         { "ro.boot.bootloader", "ro.bootloader", "unknown", },
+        { "ro.boot.boardid", "ro.boardid", "", },
     };
 
     for (i = 0; i < ARRAY_SIZE(prop_map); i++) {
